charts/consul/templates/crd-http-route-auth-filter.yaml

```diff
@@ -0,0 +1,135 @@
+{{- if .Values.connectInject.enabled }}
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: MPL-2.0
+
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    gateway.networking.k8s.io/channel: experimental
+  labels:
+    app: {{ template "consul.name" . }}
+    chart: {{ template "consul.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+    component: crd
+  name: httproute-auth-filters.consul.hashicorp.com
+spec:
+  group: consul.hashicorp.com
+  names:
+    categories:
+      - gateway-api
+    kind: HTTPRouteAuthFilter
```

Comment:
Is this name redundant/does it stutter a bit? Should we just simply rename `AuthFilter` at the CRD level?

Reply:
we could name it `AuthFilter` or maybe even `RouteAuthFilter` if it will also be applied to GRPC routes at some point in the future, I do like the `Route` prefix as it makes it a bit clearer what it applies to

Reply:
+1 to the `Route` prefix as it helps make things self-documenting and follows existing conventions w/ `Gateway`, `GatewayClass`, `GatewayClassConfig`